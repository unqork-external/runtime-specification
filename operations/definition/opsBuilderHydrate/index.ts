export * from './opsBuilderHydrate.model'
export * from './opsBuilderHydrate.options'
