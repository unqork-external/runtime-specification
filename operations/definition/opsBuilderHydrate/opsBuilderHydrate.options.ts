import { Description, Example, Required } from '@tsed/schema'

import { OpsBuilderTargets } from '../../../components/operationsBuilder/targets/operationsBuilderTargets.enum'
import { type SignalTargets } from '../../../signals'

export class OpsBuilderHydrateOperationOptions {
  @Required()
  @Example('opsBuilder')
  @Description('key of the Ops Builder component to hydrate')
  targetKey: string

  @Description('an object of root signals that resolves to an object to hydrate the ops builder component')
  declare value: SignalTargets<OpsBuilderTargets>
}
