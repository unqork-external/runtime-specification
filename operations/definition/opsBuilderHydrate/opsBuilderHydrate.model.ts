import { DiscriminatorValue, Required } from '@tsed/schema'

import { OpsBuilderHydrateOperationOptions } from './opsBuilderHydrate.options'
import { UdOnly } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.OPS_BUILDER_HYDRATE)
@UdOnly()
export class OpsBuilderHydrateOperation extends Operation<OperationTypes.OPS_BUILDER_HYDRATE> {
  @Required()
  public type = OperationTypes.OPS_BUILDER_HYDRATE as const

  @Required()
  public options: OpsBuilderHydrateOperationOptions = new OpsBuilderHydrateOperationOptions()
}
