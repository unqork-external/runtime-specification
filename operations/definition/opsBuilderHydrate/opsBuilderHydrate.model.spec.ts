import { OpsBuilderHydrateOperation } from './opsBuilderHydrate.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: OpsBuilderHydrate Tests', () => {
  it('OpsBuilderHydrate Model', () => {
    const { schema } = generateSchemaAndValidate(OpsBuilderHydrateOperation)

    expect(schema).toMatchSnapshot()
  })
})
