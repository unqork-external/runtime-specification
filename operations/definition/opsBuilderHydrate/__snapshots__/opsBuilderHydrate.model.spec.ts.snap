// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: OpsBuilderHydrate Tests OpsBuilderHydrate Model 1`] = `
{
  "definitions": {
    "OpsBuilderHydrateOperationOptions": {
      "properties": {
        "targetKey": {
          "description": "key of the Ops Builder component to hydrate",
          "examples": [
            "opsBuilder",
          ],
          "minLength": 1,
          "type": "string",
        },
        "value": {
          "description": "an object of root signals that resolves to an object to hydrate the ops builder component",
          "type": "object",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/OpsBuilderHydrateOperationOptions",
    },
    "type": {
      "const": "OPS_BUILDER_HYDRATE",
      "examples": [
        "OPS_BUILDER_HYDRATE",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "stability": "UD_ONLY",
  "type": "object",
}
`;
